src/navigator_data_ingest/base/updated_document_actions.py

```diff
@@ -271,11 +271,55 @@ def parse(
     return [error for error in errors if error is not None]
 
 
+def update_field_in_all_occurences(
+    update: Tuple[str, Update],
+    update_config: UpdateConfig,
+) -> List[Union[str, None]]:
+    """Update the document slug in all occurences of the document in s3."""
+    # TODO Do we need to archive on slug updates? The reason for this was expensive
+    # translation and text extraction costs not slug updates?
+    # TODO This can be made more generic.
+
+    document_id, document_update = update
+    _LOGGER.info(
+        "Updating document field in all document occurences in s3.",
+        extra={
+            "props": {
+                "document_id": document_id,
+            }
+        },
+    )
+    errors = []
+    for prefix in [
+        update_config.parser_input,
+        update_config.embeddings_input,
+        update_config.indexer_input,
+    ]:
+        prefix_path = S3Path(
+            os.path.join("s3://", update_config.pipeline_bucket, prefix)
+        )
```

Comment:
If you wanted to, you could also bring this logic into the list declaration in the previous comment and then loop over the full paths. (I think doing so would make this a little simpler, though I'm not sure it makes that much difference)